src/media-groups.js

```diff
@@ -269,27 +269,24 @@ export const setupListeners = {
       segmentLoaders: { [type]: segmentLoader }
     } = settings;
 
-    playlistLoader.on('loadedmetadata', () => {
+    // only start to append segments to source buffers after
+    // the main segment loader has already appended
+    settings.segmentLoaders.main.one('appended', () => {
       const media = playlistLoader.media();
 
-      segmentLoader.playlist(media, requestOptions);
-
       // if the video is already playing, or if this isn't a live video and preload
       // permits, start downloading segments
       if (!tech.paused() || (media.endList && tech.preload() !== 'none')) {
         segmentLoader.load();
       }
     });
 
-    playlistLoader.on('loadedplaylist', () => {
+    const loadPlaylist = () => {
+      playlistLoader.off(['loadedmetadata', 'loadedplaylist'], loadPlaylist);
       segmentLoader.playlist(playlistLoader.media(), requestOptions);
+    };
 
-      // If the player isn't paused, ensure that the segment loader is running
-      if (!tech.paused()) {
-        segmentLoader.load();
-      }
-    });
-
+    playlistLoader.one(['loadedmetadata', 'loadedplaylist'], loadPlaylist);
```

Comment:
this will break live

Reply:
and probably MAAT

Reply:
as in it should be `on` instead of `one`?

Reply:
yes, the `playlistloader` will trigger `loadedplaylist` every time it refreshes the live manifest. If you stop listening to it after one event, the segment loader will never get new segments.